cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -1424,7 +1429,20 @@ private void getRuns(HttpResponder responder, Id.Program programId, String statu
       try {
         ProgramRunStatus runStatus = (status == null) ? ProgramRunStatus.ALL :
           ProgramRunStatus.valueOf(status.toUpperCase());
-        responder.sendJson(HttpResponseStatus.OK, store.getRuns(programId, runStatus, start, end, limit));
+        List<RunRecord> records =
+          Lists.transform(store.getRuns(programId, runStatus, start, end, limit),
+                          new Function<RunRecordMeta, RunRecord>() {
+                            @Nullable
+                            @Override
+                            public RunRecord apply(@Nullable RunRecordMeta input) {
```

Comment:
Should not be null, but the apply contract have nullable annotation so I added the check.